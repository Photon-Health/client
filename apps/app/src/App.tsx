import { ColorModeScript } from '@chakra-ui/react'

import { Routes, Route, BrowserRouter } from 'react-router-dom'

import { AppState } from '@auth0/auth0-react'
import { PhotonClient, PhotonProvider } from '@photonhealth/react'

import { Login } from './views/routes/Login'
import { Logout } from './views/routes/Logout'
import { Main } from './views/routes/Main'
import { NotFound } from './views/routes/NotFound'
import { Orders } from './views/routes/Orders'
import { Order } from './views/routes/Order'
import { Patient } from './views/routes/Patient'
import { Prescription } from './views/routes/Prescription'
import { PatientForm } from './views/routes/PatientForm'
import { PrescriptionForm } from './views/routes/PrescriptionForm'
import { Patients } from './views/routes/Patients'
import { Prescriptions } from './views/routes/Prescriptions'
import { Settings } from './views/routes/Settings'
import { Support } from './views/routes/Support'
import { UpdatePatientForm } from './views/routes/UpdatePatientForm'
import { NewOrder } from './views/routes/NewOrder'
import { auth0Config } from './configs/auth'
import { AlertDisplay } from './views/components/AlertDisplay'

const client = new PhotonClient({
  domain: auth0Config.domain,
  clientId: auth0Config.clientId,
  redirectURI: auth0Config.redirectUri,
  audience: auth0Config.audience,
  uri: process.env.REACT_APP_GRAPHQL_URI as string
})

const onRedirectCallback = (appState?: AppState) => {
  window.location.replace(appState?.returnTo || window.location.pathname)
}

export const App = () => {
  return (
    <BrowserRouter>
      <PhotonProvider client={client} onRedirectCallback={onRedirectCallback}>
        <ColorModeScript />
        <AlertDisplay />
        <Routes>
          <Route path="/" element={<Main />}>
            <Route path="/patients">
              <Route path="/patients" element={<Patients />} />
              <Route path="new" element={<PatientForm />} />
              <Route path="update/:patientId" element={<UpdatePatientForm />} />
            </Route>
            <Route path="/patients/:patientId" element={<Patient />} />
            <Route path="/prescriptions">
              <Route path="/prescriptions" element={<Prescriptions />} />
              <Route path="new" element={<PrescriptionForm />} />
              <Route path=":prescriptionId" element={<Prescription />} />
            </Route>
            <Route path="/prescriptions/new" element={<PrescriptionForm />} />
            <Route path="/orders">
              <Route path="/orders" element={<Orders />} />
              <Route path="new" element={<NewOrder />} />
              <Route path=":orderId" element={<Order />} />
            </Route>
            <Route path="/support" element={<Support />} />
            <Route path="/settings" element={<Settings />}>
              <Route path="user" />
              <Route path="templates" />
              <Route path="catalog" />
            </Route>
          </Route>
          <Route path="/login" element={<Login />} />
          <Route path="/logout" element={<Logout />} />
          <Route path="*" element={<NotFound />} />
        </Routes>
      </PhotonProvider>
    </BrowserRouter>
  )
}
