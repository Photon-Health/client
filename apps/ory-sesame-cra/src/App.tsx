/* eslint-disable @typescript-eslint/no-unused-vars */
// import logo from "./logo.svg"
// import "./App.css"

import { Box, Button, Flex, Input } from '@chakra-ui/react';
import { PhotonClient, PhotonProvider } from '@photonhealth/react';
import { ChangeEventHandler, useRef, useState } from 'react';
import SignatureCanvas from 'react-signature-canvas';
import trimCanvas from 'trim-canvas';
import { SesamePage } from './SesameWrapper';

declare global {
  namespace JSX {
    interface IntrinsicElements {
      'photon-client': unknown;
      'photon-prescribe-workflow': unknown;
      'photon-auth-wrapper': unknown;
    }
  }
}

const domain = 'auth.boson.health';
const clientId = 'WoOY2q0gtrFuffz5WUQWlYiJ2iPGwWlp';
const redirectURI = window.location.origin;
const audience = 'https://api.boson.health';
const uri = 'https://api.boson.health/graphql';

const client = new PhotonClient({
  domain,
  clientId,
  redirectURI,
  audience,
  uri
});

const DrawCanvas = ({ setData }: { setData: (data: string) => void }) => {
  const ref = useRef<SignatureCanvas | null>(null);

  const onEnd = () => {
    const data = ref.current?.getTrimmedCanvas().toDataURL();
    if (data) {
      setData(data);
    }
  };

  return <SignatureCanvas ref={ref} onEnd={onEnd} />;
};

// return a trimmed copy of the canvas
const getTrimmedCanvas = (canvas: HTMLCanvasElement): HTMLCanvasElement => {
  // copy the canvas
  const copy = document.createElement('canvas');
  copy.width = canvas.width;
  copy.height = canvas.height;
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  copy.getContext('2d')!.drawImage(canvas, 0, 0);
  // then trim it
  return trimCanvas(copy);
};

const GenerateCanvas = ({ setData }: { setData: (data: string) => void }) => {
  const ref = useRef<HTMLCanvasElement>(null);

  const onChange: ChangeEventHandler<HTMLInputElement> = (event) => {
    if (!ref.current) return;
    const ctx = ref.current.getContext('2d');
    if (!ctx) return;
    ctx.clearRect(0, 0, ref.current.width, ref.current.height);
    ctx.font = '48px cursive';

    ctx.textBaseline = 'hanging';
    ctx.fillText(event.target.value, 0, 12);
    setData(getTrimmedCanvas(ref.current).toDataURL());
  };

  return (
    <div>
      <canvas ref={ref} width="500" height="100" />
      <Input onChange={onChange} />
    </div>
  );
};

const SignatureComponent = () => {
  const [isEditing, setIsEditing] = useState(true);
  const [useDraw, setUseDraw] = useState(false);
  const [data, setInnerData] = useState<string>('');
  const [date, setDate] = useState<string>('');

  const setData = (data: string) => {
    setInnerData(data);
    setDate(new Date().toLocaleString());
  };

  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        width: '100%',
        height: '100%',
        border: '1px solid black'
      }}
    >
      {isEditing && (
        <div style={{ display: 'flex', width: '100%' }}>
          <Button onClick={() => setUseDraw(false)}>Generate</Button>
          <Button onClick={() => setUseDraw(true)}>Draw</Button>
        </div>
      )}
      {isEditing ? (
        useDraw ? (
          <DrawCanvas setData={setData} />
        ) : (
          <GenerateCanvas setData={setData} />
        )
      ) : (
        <div>
          <img src={data} />
          <div>{date}</div>
        </div>
      )}
      <Button
        onClick={() => {
          if (!isEditing) {
            setData('');
          }
          setIsEditing(!isEditing);
        }}
      >
        {isEditing ? 'Save' : 'Edit'}
      </Button>
    </div>
  );
};

function App() {
  return (
    <div className="App">
      <Flex w="100vw" h="100vh" justifyContent={'center'} p="50">
        <SignatureComponent />
      </Flex>
      {/* <PhotonProvider client={client}>
        <SesamePage>
          <Flex pr="10" h="100%" flexGrow={'1'}>
            <Box width="20%" bg={'green'}></Box>
            <Box flex="1">
              <photon-client
                id={clientId}
                org="org_qzo8REINlFXB5v24"
                domain={domain}
                audience={audience}
                uri={uri}
                auto-login="true"
              >
                <photon-prescribe-workflow />
              </photon-client>
            </Box>
          </Flex>
        </SesamePage>
      </PhotonProvider> */}
    </div>
  );
}

export default App;
