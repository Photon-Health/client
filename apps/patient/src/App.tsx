import { Routes, Route, BrowserRouter } from 'react-router-dom'

import { Main } from './views/Main'
import { Status } from './views/Status'
import { NoMatch } from './views/NoMatch'
import { Pharmacy } from './views/Pharmacy'
import { Review } from './views/Review'
import { ReadyBy } from './views/ReadyBy'

export const App = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Main />}>
          <Route path="/ready" element={<ReadyBy />} />
          <Route path="/review" element={<Review />} />
          <Route path="/pharmacy" element={<Pharmacy />} />
          <Route path="/status" element={<Status />} />
        </Route>
        <Route path="*" element={<NoMatch />} />
      </Routes>
    </BrowserRouter>
  )
}
