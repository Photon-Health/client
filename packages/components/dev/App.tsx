import { For, createMemo, createSignal, createEffect } from 'solid-js';
import PharmacySearch from '../src/systems/PharmacySearch';
import DoseCalculator from '../src/systems/DoseCalculator';
import Client from '../src/systems/Client';
import Button from '../src/particles/Button';
import Icon from '../src/particles/Icon';
import { randomNames } from '../src/sampleData/randomNames';
import ComboBox from '../src/particles/ComboBox';
import PharmacySelect from '../src/systems/PharmacySelect';
import Card from '../src/particles/Card';

const App = () => {
  const [setPharmacy] = createSignal<any>();
  const [doseOpen, setDoseOpen] = createSignal(false);
  const [query, setQuery] = createSignal('');
  const [patientIds, setPatientIds] = createSignal<string[]>([]);
  const rando = randomNames.slice(0, 3);
  const filteredPeople = createMemo(() => {
    return query() === ''
      ? rando
      : rando.filter((person) => {
          return person.name.toLowerCase().includes(query().toLowerCase());
        });
  });

  createEffect(() => {
    setTimeout(() => {
      setPatientIds(['pat_01H28NXFX27PSADPYPR5JHTCD7']);
    }, 2000);
  });

  return (
    <div class="sm:max-w-lg p-4">
      <div>
        <h1>Photon Component Systems</h1>
      </div>
      <Client
        id="7N9QZujlNJHL8EIPqXpu1wq8OuXqoxKb"
        org="org_KzSVZBQixLRkqj5d"
        domain="auth.boson.health"
        audience="https://api.boson.health"
        uri="https://api.boson.health/graphql"
      >
        <div class="mb-10">
          <div>
            <h2>ComboBox</h2>
            <ComboBox>
              <ComboBox.Input
                onInput={(e) => setQuery(e.currentTarget.value)}
                displayValue={(person) => person.name}
              />
              <ComboBox.Options>
                <For each={filteredPeople()}>
                  {(person) => (
                    <ComboBox.Option key={person.id} value={person}>
                      {person.name}
                    </ComboBox.Option>
                  )}
                </For>
              </ComboBox.Options>
            </ComboBox>
          </div>
        </div>

        <div class="mb-10">
          <div>
            <h2>Pharmacy Search</h2>
            <PharmacySearch setPharmacy={setPharmacy} />
          </div>
        </div>

        <div class="mb-10">
          <div>
            <h2>Dose Calculator</h2>
            <Button onClick={() => setDoseOpen(true)}>
              <Icon name="calculator" size="sm" />
            </Button>

            <DoseCalculator
              open={doseOpen()}
              onClose={() => setDoseOpen(false)}
              setAutocompleteValues={({ liquidDose, totalLiquid, unit }) => {
                console.log(liquidDose, totalLiquid, unit);
              }}
            />
          </div>
        </div>

        <h2>Pharmacy Select</h2>
        <Card>
          <PharmacySelect
            patientIds={patientIds()}
            displayLocalPickup
            displaySendToPatient
            setFufillmentType={(t) => console.log('fulfillmentType: ', t)}
            setPharmacyId={(p) => console.log('pharmacyId: ', p)}
          />
        </Card>
        <h4 class="mt-8">With Mail Order</h4>
        <Card>
          <PharmacySelect
            patientIds={['pat_01H28NXFX27PSADPYPR5JHTCD7']}
            displayLocalPickup
            displaySendToPatient
            mailOrderPharmacyIds={[
              'phr_01GA9HPVBVJ0E65P819FD881N0',
              'phr_01GCA54GVKA06C905DETQ9SY98'
            ]}
            setPharmacyId={(p) => console.log('pharmacyId! ', p)}
            setFufillmentType={(t) => console.log('fulfillmentType! ', t)}
          />
        </Card>
      </Client>
    </div>
  );
};

export default App;
