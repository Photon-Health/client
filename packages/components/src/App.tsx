import { createSignal } from 'solid-js';
import PharmacySearch from './systems/PharmacySearch';
import Client from './systems/Client';
import SelectLocation from './systems/SelectLocation';

const App = () => {
  const [pharmacy, setPharmacy] = createSignal<any>();

  return (
    <Client
      id="7N9QZujlNJHL8EIPqXpu1wq8OuXqoxKb"
      org="org_KzSVZBQixLRkqj5d"
      domain="auth.boson.health"
      audience="https://api.boson.health"
      uri="https://api.boson.health/graphql"
    >
      <div class="p-5" style={{ width: '600px' }}>
        <div class="p-5 border border-gray-300 rounded-md">
          <PharmacySearch setPharmacy={setPharmacy} />
        </div>
      </div>

      <div class="p-5" style={{ width: '600px' }}>
        <div class="p-5 border border-gray-300 rounded-md">
          <p class="pb-5">SelectLocation Component</p>
          <SelectLocation />
        </div>
      </div>
    </Client>
  );
};

export default App;
