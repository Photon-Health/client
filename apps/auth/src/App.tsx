import './App.css';
import LoginButton from './LoginButton';
import { useAuth0 } from '@auth0/auth0-react';
import LogoutButton from './LogoutButton';

function App() {
  const { user, isAuthenticated, isLoading } = useAuth0();

  if (isLoading) {
    return <div>Loading ...</div>;
  }

  return (
    <>
      {!isAuthenticated ? (
        <LoginButton />
      ) : (
        <photon-client
          id="7N9QZujlNJHL8EIPqXpu1wq8OuXqoxKb"
          org="org_KzSVZBQixLRkqj5d"
          domain="auth.boson.health"
          env="boson"
          audience="https://api.boson.health"
          uri="https://api.boson.health/graphql"
          auto-login="true"
          redirect-uri={`${window.location.origin}?photon-child=true`}
        >
          <div>
            <LogoutButton />
            <div>
              <img src={user.picture} alt={user.name} />
              <h2>{user.name}</h2>
              <p>{user.email}</p>
            </div>
            <photon-prescribe-workflow
              patient-id="pat_01GQ0XFBHSH3YXN936A2D2SD7Y"
              enable-order="true"
              enable-send-to-patient="true"
              enable-local-pickup="true"
              enable-med-history="true"
              mail-order-ids="phr_01GA9HPVBVJ0E65P819FD881N0,phr_01GCA54GVKA06C905DETQ9SY98"
              additional-notes="Some additional notes to add to the message."
              weight="23"
            ></photon-prescribe-workflow>
          </div>
        </photon-client>
      )}
    </>
  );
}

export default App;
