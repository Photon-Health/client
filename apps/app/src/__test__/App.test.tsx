import { create } from 'react-test-renderer'
import { App } from '../App'

describe('App', () => {
  test('renders correctly', () => {
    const component = create(<App />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
