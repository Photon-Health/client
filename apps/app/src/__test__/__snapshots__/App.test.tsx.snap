// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <script
    dangerouslySetInnerHTML={
      Object {
        "__html": "!(function(){try{var a=function(c){var v=\\"(prefers-color-scheme: dark)\\",h=window.matchMedia(v).matches?\\"dark\\":\\"light\\",r=c===\\"system\\"?h:c,o=document.documentElement,s=document.body,l=\\"chakra-ui-light\\",d=\\"chakra-ui-dark\\",i=r===\\"dark\\";return s.classList.add(i?d:l),s.classList.remove(i?l:d),o.style.colorScheme=r,o.dataset.theme=r,r},n=a,m=\\"light\\",e=\\"chakra-ui-color-mode\\",t=localStorage.getItem(e);t?a(t):localStorage.setItem(e,a(m))}catch(a){}})();",
      }
    }
    id="chakra-script"
  />,
  <div
    className="css-h74a88"
  >
    <div
      aria-valuemax={100}
      aria-valuemin={0}
      className="chakra-progress css-120wkjd"
      data-indeterminate=""
      role="progressbar"
    >
      <svg
        className="css-jf70f3"
        viewBox="0 0 100 100"
      >
        <circle
          className="chakra-progress__track css-1bhxzgy"
          cx={50}
          cy={50}
          r={42}
          strokeWidth="10px"
        />
        <circle
          className="chakra-progress__indicator css-1b60oz0"
          cx={50}
          cy={50}
          r={42}
          strokeWidth="10px"
        />
      </svg>
    </div>
  </div>,
]
`;
